shopfloor/actions/search.py

```diff
@@ -46,11 +46,17 @@ def product_from_scan(self, barcode):
             product = packaging.product_id
         return product
 
-    def lot_from_scan(self, barcode):
+    def lot_from_scan(self, barcode, products=None, limit=1):
         model = self.env["stock.production.lot"]
         if not barcode:
             return model.browse()
-        return model.search([("name", "=", barcode)], limit=1)
+        domain = [
+            ("company_id", "=", self.env.company.id),
```

Comment:
I don't know what is the best option: filtering on current user's company or on the picking type company.